Saule/Http/JsonApiMediaTypeFormatter.cs

```diff
@@ -110,6 +110,13 @@ public override bool CanWriteType(Type type)
                 preprocessed = PreprocessingDelegatingHandler.PreprocessRequest(value, _request, _config);
             }
 
+            var preSerializedJson = preprocessed.ResourceSerializer.Value as JObject;
```

Comment:
Maybe this code should be in the `PreprocessingDelegatingHandler`? That way it also works from `JsonApiSerializer`. I know in your specific use-case that doesn't make any difference, but I've tried to keep the serializer and formatter in sync feature-wise.